### PR TITLE
Simplify if statements

### PR DESCRIPTION
**Pull request by SaberMaycry**
When I read the source code with IDEA, I found that some  statements can be simplified, so I submitted this pull request.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@SaberMaycry Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=17785)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-boot&pullRequestId=17785) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@SaberMaycry Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=17785)!

File: spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/main/java/org/springframework/boot/gradle/tasks/bundling/LaunchScriptConfiguration.java [hunk 1]
```diff
@@ -106,17 +106,12 @@ public boolean equals(Object obj) {
...
+			return other.script == null;
 		}
 		else if (!this.script.equals(other.script)) {
 			return false;
 		}
-		else if (!equalContents(this.script, other.script)) {
-			return false;
-		}
-		return true;
+		else return equalContents(this.script, other.script);
```
Comment (mbhave) [inline spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/main/java/org/springframework/boot/gradle/tasks/bundling/LaunchScriptConfiguration.java]: @SaberMaycry Can you fix the checkstyle error on this line please?

Comment (SaberMaycry) [reply spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/main/java/org/springframework/boot/gradle/tasks/bundling/LaunchScriptConfiguration.java]:
```suggestion
		else return equalContents(this.script, other.script);
```
Of course.

**Comment by snicoll** [comment]
@SaberMaycry thank you for making your first contribution to Spring Boot. I've decided to revert a couple of changes when there are several if statements in a row as I prefer the readability even if the last statement can be simplified. Thanks again.

**Pull request closed without merging**